### PR TITLE
Supplier name validation fix

### PR DESCRIPTION
**Pull request by KatyTaylor**
don't apply the validation to supplier name if you're blanking it

### CODE REVIEW

File: spec/models/sample_spec.rb [hunk 1]
```diff
@@ -149,4 +149,22 @@
...
+    it 'validates that supplier name allows only ASCII characteres' do
+      expect(sample.sample_metadata.supplier_name).to be_nil
+      sample.sample_metadata.supplier_name = 'भारत'
+      expect(sample.sample_metadata.save).to eq false
+    end
+
+    it 'can have the supplier name blanked' do
+      expect(sample.sample_metadata.supplier_name).to be_nil
+      sample.sample_metadata.update!(supplier_name: 'something')
+      expect(sample.sample_metadata.supplier_name).not_to be_nil
```
Comment (harrietc52) [inline spec/models/sample_spec.rb:165]: Would save eq true be more useful than `update!`?

File: spec/models/sample_spec.rb [hunk 2]
```diff
@@ -149,4 +149,22 @@
...
+      sample.sample_metadata.update!(supplier_name: nil)
+      expect(sample.sample_metadata.supplier_name).to be_nil
```
Comment (harrietc52) [inline spec/models/sample_spec.rb:167]: And save eq true here too?
Is `null` acting the same as `nil`?

File: app/models/sample.rb [hunk 3]
```diff
@@ -230,7 +230,7 @@ class Metadata
                 with: /\A[[:ascii:]]+\z/,
                 message: 'only allows ASCII'
               },
-              if: :supplier_name_changed?
+              if: :supplier_name_changed? && :supplier_name?
```
Comment (harrietc52) [inline app/models/sample.rb:233]: Does this automatiicaly trigger the `touch` which resends the sample to the MLWH?

**Review by harrietc52 — APPROVED** [review]
Looks good 👍 few minor questions

File: app/models/sample.rb (hunk 3 above)
Comment (KatyTaylor) [reply app/models/sample.rb:233]: Hopefully it should be sent to the warehouse when we update to nil. Didn't test it locally though so I will def check this in UAT.

File: spec/models/sample_spec.rb (hunk 2 above)
Comment (KatyTaylor) [reply spec/models/sample_spec.rb:167]: `nil` is just Ruby for `null`

File: spec/models/sample_spec.rb (hunk 1 above)
Comment (KatyTaylor) [reply spec/models/sample_spec.rb:165]: Not sure, what else would it tell us?

File: app/models/sample.rb (hunk 3 above)
Comment (KatyTaylor) [reply app/models/sample.rb:233]: Tested this in UAT - updates MLWH fine when filling in and blanking supplier_name

**Pull request merged**